tests/conftest.py

```diff
@@ -150,6 +152,38 @@ def pytest_addoption(parser):  # type: ignore
         default=False,
         help="enable integration tests",
     )
+    parser.addoption(
+        "--error-on-pending-deprecations",
+        action="store_true",
+        default=False,
+        help="Treat PendingDeprecationWarnings from anaconda_auth as errors",
+    )
+
+
+def pytest_configure(config):
+    """Customize warning filters to ignore PendingDeprecationWarnings that are
+    raised by other conda plugins, while allowing us to still catch our own.
+    """
+    if config.getoption("--error-on-pending-deprecations"):
+        warnings.simplefilter("error", PendingDeprecationWarning)
+
+        original_warn = warnings.warn
+
+        def custom_warn(message, category=UserWarning, stacklevel=1, source=None):
+            if isinstance(category, type) and issubclass(
+                category, PendingDeprecationWarning
+            ):
+                stack = traceback.extract_stack()
+                # Ignore the warning if any of the known plugins matches
+                for frame in stack:
+                    if "conda_libmamba_solver" in frame.filename:
+                        return
+                    if "anaconda_anon_usage" in frame.filename:
+                        return
```

Comment:
No others? Should it be inclusive rather than exclusive?

Reply:
Well, if it works. It feels like there's probably a library out there for something like this, but you wouldn't want another dep even for testing.